#166272299 users view edit profile

adex001 opened the pull request:
#### What does this PR do?
- implement update and view profile

#### Description of Task to be completed?
- setup update-profile component
- set up view profile component
- implement container to view and update the store

#### How should this be manually tested?
clone this repo
run `npm install`
run `npm start`
run `npm run test`
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#166272299](https://www.pivotaltracker.com/story/show/166272299)
#### Screenshots (if appropriate)
<img width="1440" alt="Screenshot 2019-07-19 at 3 14 51 PM" src="https://user-images.githubusercontent.com/29796551/61542150-e3e77c80-aa38-11e9-818e-f5b6504ed701.png">
<img width="1440" alt="Screenshot 2019-07-19 at 3 17 03 PM" src="https://user-images.githubusercontent.com/29796551/61542179-eea21180-aa38-11e9-9dc3-421be4356b0d.png">


#### Questions:
N/A

adex001 committed:
feat-[166272299]: users view and edit profile

- designed the skeleton for view and edit profile

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- update prop types validations

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- update image upload process

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- update save functionality to database

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

-fix dotenv issue

[Delivers #166272299]

fidelisojeah commented on src/components/views/Profile/Profile.jsx:
```diff
@@ -0,0 +1,83 @@
+  }
+  render() {
+    const { user, profile, isAuthenticated } = this.props;
+    const {tabs, clickedTab} = this.state;
```
Why was this not flagged by your local eslint?

adex001 replied:
I guess its properly destructured.

adex001 committed:
feat-[166272299]: users view and edit profile

- implement feedback and made route to redirect when user not found

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- implement feedback and configure webpack to accept .env variables

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- update test for profile

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- create a isLoggedIn component to verify if logged in

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- implement users page and make edit button to show upon authentication

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- implement users page and make edit button to show upon authentication

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- update responsiveness of the edit profile container

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

implement feedback form reviewers

[Delivers #166272299]

fidelisojeah commented on src/components/container/Profile/Profile.jsx:
```diff
@@ -0,0 +1,103 @@
+  componentDidMount() {
+    const { user: {username}, fetchGuest, history  } = this.props;
+    const { match: { params: { username: usernameURL } } } = this.props;
+    if (username === usernameURL) {
```
You could probably do this in the constructor.

`isMyProfile: this.props.match.params.username === this.props.user.username`

adex001 replied:
this is already implemented. Thanks!

adex001 committed:
feat-[166272299]: users view and edit profile

implement feedback from Fidelis

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

implement feedback from Fidelis

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- resolve merge conflict

[Delivers #166272299]

adex001 committed:
feat-[166272299]: users view and edit profile

- resolve merge conflict and implement feedback

[Delivers #166272299]
